### PR TITLE
ISSUE-8992 Fix normal-state keymap is not restored issue

### PR DESCRIPTION
**Pull request by enzochiau**
As reported in #8992, after exit `evilified-state`, the normal-state keymap is not restored.
This fix adds restore function to exit-hook.

### CODE REVIEW

File: layers/+distributions/spacemacs-base/local/evil-evilified-state/evil-evilified-state.el [hunk 1]
```diff
@@ -60,7 +60,8 @@
   :tag " <N'> "
   :enable (emacs)
   :message "-- EVILIFIED BUFFER --"
-  :cursor box)
+  :cursor box
+  :exit-hook (list evilified-state--restore-normal-state-keymap))
```
Comment (TheBB) [inline layers/+distributions/spacemacs-base/local/evil-evilified-state/evil-evilified-state.el]: You shouldn't have `list` here.

**Comment by TheBB** [comment]
Yes, but `evil-define-state` is a macro, all its arguments are quoted, and this one is not evaluated. So the final hook you get has two functions, one of them is `list`.

**Comment by enzochiau** [comment]
Thanks for the explanation. "list" is removed.

**Comment by Miciah** [comment]
This change is similar to #7767, which was reverted because it broke mu4e.  Has anyone tested this change with mu4e?

**Comment by enzochiau** [comment]
After almost 3 years, I still have this issue when exit from `edebug-mode`. I am not sure if the fix still break the `mu4e`.

**Pull request closed without merging**